Bluetooth: audio: ascs: Fix handling zero Number_of_ASEs parameter

MariuszSkamra opened the pull request:
As per ASCS_v1.0
"A client-initiated ASE Control operation shall be defined as an
invalid length operation if the Number_of_ASEs parameter value is
less than 1".

Fixes: ASCS/SR/SPE/BI-{02,03,04,05,06}-C
Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra committed:
Bluetooth: audio: ascs: Fix handling zero Number_of_ASEs parameter

As per ASCS_v1.0
"A client-initiated ASE Control operation shall be defined as an
invalid length operation if the Number_of_ASEs parameter value is
less than 1".

Fixes: ASCS/SR/SPE/BI-{02,03,04,05,06}-C
Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra commented:
@szymon-czapracki FYI

jhedberg commented on subsys/bluetooth/audio/ascs.c:
```diff
@@ -1312,7 +1312,10 @@ static ssize_t ascs_config(struct bt_ascs *ascs, struct net_buf_simple *buf)
 
-	if (buf->len < req->num_ases * sizeof(*cfg)) {
+	if (req->num_ases < 1) {
+		BT_ERR("Number_of_ASEs parameter value is less than 1");
```
I realize that the issue already existed in this file, but BT_ERR() should really be reserved for local errors only. Whenever a remote peer sends something invalid that we cleanly handle then BT_WARN() is the right choice.

jhedberg commented:
@MariuszSkamra do you want to update the `else if` branches as well (perhaps a separate commit in this PR)? You now have a mix of BT_WARN/BT_ERR which is not ideal..

MariuszSkamra committed:
Bluetooth: audio: ascs: Use BT_WARN to log peer invalid behavior

BT_ERR() should really be reserved for local errors only,
thus BT_WARN() is the right choice whenever a remote peer
sends something invalid that we cleanly handle.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra commented:
> @MariuszSkamra do you want to update the `else if` branches as well (perhaps a separate commit in this PR)? You now have a mix of BT_WARN/BT_ERR which is not ideal..

I've pushed it as separate commit
